### DISCUSSION TITLE
Github Action cloning parent of fork repository when initiated from child fork

### DISCUSSION THREAD

**Discussion by mishraprafful** [community/community#27057] [Actions]
<p>Github action step trying to run a bash script that calculates git diff has default work directory as the remote parent fork repository.</p>
<p>This GitHub Action was initiated from a fork of the mentioned repository.</p>
<p><img src="https://user-images.githubusercontent.com/32913433/181101525-1dd82764-2762-4133-85fd-d1cc895a7ed8.png" alt="Screenshot 2021-03-04 at 11.43.14" data-base62-sha1="6M81z17VSXWQWV8hMx3olGJYszh" width="623" height="85"></p>

**Comment by Simran-B** [comment, marked as answer]
<p>Can you share the workflow code and describe the setup in more detail? This is too little information to help.</p>
<blockquote>
<p>[…] has default work directory as the remote parent fork repository</p>
</blockquote>
<p>This sounds like you might use <code>pull_request_target</code> which will checkout the base branch instead of the head branch for security reasons (to prevent arbitrary users to be able to modify your workflow to leak secrets).</p>

**Comment by mishraprafful** [comment]
<p>I think this answers my question <a class="mention" href="/simran-b">@Simran-B</a>. Thanks for your response.</p>

**Comment by Simran-B** [comment]
<p>In case you want to learn more about <code>pull_request_target</code> and security:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://securitylab.github.com/research/github-actions-preventing-pwn-requests/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/7819991/181101529-ad50411c-13ed-4ec6-8f43-05ee79c286e0.png" class="site-icon" width="72" height="72">

      <a href="https://securitylab.github.com/research/github-actions-preventing-pwn-requests/" target="_blank" rel="noopener nofollow ugc" title="12:00AM - 15 December 2020">GitHub Security Lab – 15 Dec 20</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/7819991/181101530-26c37b27-205c-484a-b8fe-9f815fe85a28.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/c/ec60f25ad72688253445273a7394448fe64ea35a_2_10x10.png"></div>

<h3><a href="https://securitylab.github.com/research/github-actions-preventing-pwn-requests/" target="_blank" rel="noopener nofollow ugc">Keeping your GitHub Actions and workflows secure: Preventing pwn requests</a></h3>

  <p>In this article, we’ll discuss some common security malpractices for GitHub Actions and workflows, and how to best avoid them. Our examples are based on real-world GitHub workflow implementation vulnerabilities the GitHub Security Lab has reported to...</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
